### PR TITLE
Speed up tests execution

### PR DESCRIPTION
**Pull request by marek-safar**
This changes reduced total tests execution time under netcore from about
15 minutes on local or CI machine to about 3 minutes.

### CODE REVIEW

File: test/Mono.Linker.Tests/TestCasesRunner/LinkerArgumentBuilder.cs [hunk 1]
```diff
@@ -184,6 +184,8 @@ public virtual void ProcessOptions (TestCaseLinkerOptions options)
 		{
 			if (options.CoreAssembliesAction != null)
 				AddCoreLink (options.CoreAssembliesAction);
+			else
+				AddCoreLink ("skip");
```
Comment (vitek-karas) [inline test/Mono.Linker.Tests/TestCasesRunner/LinkerArgumentBuilder.cs:188]: This feels a bit off - how would we test the default behavior of the linker (That is omit the parameter)?
Not saying this it needs to be easy, but it feels wrong that we obviously don't test that behavior.

File: test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs [hunk 2]
```diff
@@ -169,6 +169,19 @@ protected static NPath MakeSupportingAssemblyReferencePathAbsolute (NPath output
 			if (Path.IsPathRooted (referenceFileName))
 				return referenceFileName.ToNPath ();
 
+			if (referenceFileName.StartsWith ("System.", StringComparison.Ordinal) ||
+				referenceFileName.StartsWith ("Mono.", StringComparison.Ordinal) ||
+				referenceFileName.StartsWith ("Microsoft.", StringComparison.Ordinal) ||
+				referenceFileName == "netstandard.dll") {
+#if NETCOREAPP
```
Comment (vitek-karas) [inline test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs:176]: Put the entire if in the ifdef.

**Review by vitek-karas — APPROVED** [review]
Mostly minor comments - looks good!

File: test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs (hunk 2 above)
Comment (marek-safar) [reply test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs]: Changed

File: test/Mono.Linker.Tests/TestCasesRunner/LinkerArgumentBuilder.cs (hunk 1 above)
Comment (marek-safar) [reply test/Mono.Linker.Tests/TestCasesRunner/LinkerArgumentBuilder.cs:188]: I think such a test should be part of the integration like tests (basically different setup). This setup is primary about IL/metadata logic not about default behaviour.

File: src/linker/Linker.Steps/DynamicDependencyLookupStep.cs [hunk 3]
```diff
@@ -15,6 +15,9 @@ public class DynamicDependencyLookupStep : LoadReferencesStep
 	{
 		protected override void ProcessAssembly (AssemblyDefinition assembly)
 		{
+			if (Annotations.GetAction (assembly) == AssemblyAction.Skip)
```
[the remark this answers is not in the document]
Comment (marek-safar) [reply src/linker/Linker.Steps/DynamicDependencyLookupStep.cs:18]: I see that more as a bug fix (we should not process assembly if it's explicitly skipped). We could backport that to 5 where it was introduced if you think it's significant.

[the remark this answers is not in the document]
Comment (vitek-karas) [reply src/linker/Linker.Steps/DynamicDependencyLookupStep.cs:18]: I don't think it's significant enough to meet the bar for 5.0 at this time.
I agree that the new behavior is better - my question was more about our thinking about intentional behavioral changes (like this one). In short - similar change in runtime would be discussed in a more detail to determine potential impact of breaking something.

**Pull request merged**